agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryChannel.java

```diff
@@ -121,6 +123,10 @@ public CompletableResultCode sendRawBytes(
       String instrumentationKey,
       Runnable onSuccess,
       Consumer<Boolean> onFailure) {
+    if (isStatsbeat && statsbeatHasBeenShutdown.get()) {
```

Comment:
statsbeathasBeenshutdown.get() will never get called since statsbeat has been shutdown.

Reply:
i.e. sendRawByte will never get called again after the shut down.

Reply:
what about retries from disk?

Reply:
can we make shutdown handle that?

Reply:
I don't understand, can you explain more what you mean?

Reply:
when we call statsbeatModule.shutdown(), that method will stop the job scheduler + stop the loader statsbeat telemetry from disk?

Reply:
or just stop the job scheduler for LocalFileLoader for statsbeat. I think there is a shutdown method too..

Reply:
ah I understand, that's a good idea, not super easy to implement, and will cause (even more) conflicts, are you ok to revisit this in vnext branch after I merge these changes forward?

Reply:
sure, let's add a todo we can change it in vnext.

Reply:
TODO added